### PR TITLE
Super Staff Bros.: Add %Bumbadadabum

### PR DESCRIPTION
**Pull request by RhythmsAreFromVibrations**

### CODE REVIEW

**Comment by RhythmsAreFromVibrations** [comment]
Spoke with him over Smogon about this

File: config/formats.js [hunk 1]
```diff
@@ -3431,6 +3431,20 @@ exports.Formats = [
 					this.add('-anim', source, "Tail Slap", target);
 				};
 			}
+			if (move.id === 'hypervoice' && name === 'bumbadadabum') {
+				move.name = 'Open Source Software';
+				move.type = 'Electric';
+				move.basePower = 110;
+				move.accuracy = 95;
+				move.secondaries = [{chance:20,self:{boosts:{spa:-1}},volatileStatus:'disable'}];
```
Comment (ascriptmaster) [inline config/formats.js]: `Error: Operator , should not stick to following expression at formats.js`

Add a space after every comma.

Comment (RhythmsAreFromVibrations) [inline config/formats.js]: yeah i was doing that ;-;

**Comment by RhythmsAreFromVibrations** [comment]
Fixed spacing errors

File: config/formats.js (hunk 1 above)
Comment (Zarel) [inline config/formats.js]: Can you do that sort of thing before making the pull request, instead of after?

File: config/formats.js [hunk 2]
```diff
@@ -3431,6 +3431,20 @@ exports.Formats = [
...
+				move.type = 'Electric';
+				move.basePower = 110;
+				move.accuracy = 95;
+				move.secondaries = [{chance:20, self:{boosts:{spa:-1}}, volatileStatus:'disable'}];
+				move.onTryHit = function (target, source) {
+					this.attrLastMove('[still]');
+					this.add('-anim', source, "Dark Void", target);
+				};
+				move.onHit = function () {
+					this.add('c|%Bumbadadabum|I\'d just like to interject for a moment. What you\'re referring to as Linux, is in fact, GNU/Linux, or as I\'ve recently taken to calling it, GNU plus Linux. Linux is not an operating system unto itself, but rather another free component of a fully functioning GNU system made useful by the GNU corelibs, shell utilities and vital system components comprising a full OS as defined by POSIX.');
```
Comment (xfix) [inline config/formats.js]: @bumbadadabum (you are totally a moderator now, nah, I just refer to you on GitHub)

I find it interesting that you call it GNU/Linux ([how it should be called according to GNU](https://www.gnu.org/gnu/why-gnu-linux.html)). At the same time, your signature move name "Open Source Software" ([when according to GNU it should have been Free Software](https://www.gnu.org/philosophy/open-source-misses-the-point.html)). Not really consistent.

Although, I guess it's just a political issue. Obviously nothing affecting merging this pull request.

File: data/scripts.js [hunk 3]
```diff
@@ -3669,10 +3669,16 @@ exports.BattleScripts = {
...
+				species: 'Samurott', ability: 'Analytic', item: 'Safety Goggles', gender: 'M',
+				moves: ['calmmind', 'originpulse', 'icebeam'],
+				baseSignatureMove: 'hypervoice', signatureMove: "Open Source Software",
+				evs: {hp:252, spa:252, spd:4}, nature: 'Modest'
+			},
 			'%Charles Carmichael': {
 				species: 'Quagsire', ability: 'Sap Sipper', item: 'Liechi Berry', gender: 'M',
 				moves: ['waterfall', 'earthquake', ['stoneedge', 'rockslide'][this.random(2)], 'icepunch'],
-				baseSignatureMove: 'swagger', signatureMove: 'Bad Pun',
+				baseSignatureMove: 'swagger', signatureMove: "Bad Pun",
```
Comment (kotarou3) [inline data/scripts.js]: These changes should be in a separate commit

Comment (Joimer) [inline data/scripts.js]: Yep, kota is right on this. Can you do it?

File: config/formats.js (hunk 2 above)
Comment (ScottehMax) [inline config/formats.js]: It's a [copypasta](https://wiki.installgentoo.com/index.php/Interjection), @xfix.

File: data/scripts.js (hunk 3 above)
Comment (RhythmsAreFromVibrations) [inline data/scripts.js]: Yeah I'll do it in a bit

**Comment by RhythmsAreFromVibrations** [comment]
Did

**Comment by RhythmsAreFromVibrations** [comment]
YES IT FINALLY SYNCED

I mean, I made the separate commit @kotarou3 @Joimer

File: config/formats.js [hunk 4]
```diff
@@ -3431,6 +3431,20 @@ exports.Formats = [
 					this.add('-anim', source, "Tail Slap", target);
 				};
 			}
+			if (move.id === 'hypervoice' && name === 'bumbadadabum') {
+				move.name = 'Open Source Software';
+				move.type = 'Electric';
+				move.basePower = 110;
+				move.accuracy = 95;
+				move.secondaries = [{chance:20, self:{boosts:{spa:-1}}, volatileStatus:'disable'}];
```
Comment (Joimer) [inline config/formats.js]: Is the point of this to have a 20% chance to inflict disable and -1 spa to self?

Comment (RhythmsAreFromVibrations) [inline config/formats.js]: yes

Comment (RhythmsAreFromVibrations) [inline config/formats.js]: it's because of the basePower, so whenever the secondary is active, it lowers Bumbadadabum's SpA

Comment (RhythmsAreFromVibrations) [inline config/formats.js]: rhythms: https://github.com/Zarel/Pokemon-Showdown/pull/1797
rhythms: does this look good for your move?
rhythms: i made the disabling chance to be 20% only
rhythms: same with the drop
%Bumbadadabum: that's fine

Comment (codelegend) [inline config/formats.js]: as hypervoice has no secondaries, why can't the custom secondary effect just be `move.secondary = {chance:20, self:{boosts:{spa:-1}}, volatileStatus:'disable'}`

Comment (ascriptmaster) [inline config/formats.js]: You can't. The engine only checks `move.secondaries` to figure out secondary chances. If `move.secondary` is defined, `Tools#getMove` sets `move.secondaries = [move.secondary]` (approximately) and at this point in the code, you're modifying a copy of the data retrieved by `getMove`.

Comment (RhythmsAreFromVibrations) [inline config/formats.js]: what he said

Comment (codelegend) [inline config/formats.js]: oh ok.
